tech-project-back-end/Services/IService/IUserService.cs

```diff
@@ -1,7 +1,20 @@
-﻿namespace tech_project_back_end.Services.IService
+﻿using tech_project_back_end.DTO;
+using tech_project_back_end.DTO.Users;
+using tech_project_back_end.Models;
+
+namespace tech_project_back_end.Services.IService
 {
     public interface IUserService
     {
         public Task<int> GetTotalUser();
+        public Task<List<UserDTO>> GetAllUser();
+        public Task<UserDTO> GetUserById(string userid);
+        public Task<(bool Success, string? Message, UserDTO? User, string? Token)> Login(UserLoginDTO userLogin);
+        public Task<(bool Success, string? Message, UserDTO? User)> UpdateUser(UserUpdateDTO updatedUser);
+        public Task<(bool Success, string? Message)> ForgetPassword(string email);
+        public Task<(bool Success, string? Message, UserDTO? User, string? Token)> Register(UserDTO userRegisterDto);
+        public Task<string> CreateToken(User user);
+        public string GetEmailTemplate(string templateName);
```

Comment:
Is this method using in another component?. If not. Please dont defined in IService

Reply:
For the two GetEmailTemplate and ReplacePlaceholders methods, moving them EmailService while CreateToken remains the same. Or do you propose creating a Helper class in Utilities to handle miscellaneous tasks like CreateToken?

Reply:
my main idea is if this method is only using inside its service. you shouldnt define it int Iservice